api/tests/bot_server_test_lib.py

```diff
@@ -0,0 +1,39 @@
+from unittest import TestCase
+import zulip.bot_server
+import json
+from typing import Any, List, Dict, Mapping
+
+class BotServerTestCase(TestCase):
+
+    def setUp(self):
+        # type: () -> None
+        zulip.bot_server.app.testing = True
+        self.app = zulip.bot_server.app.test_client()
+
+    def assert_bot_server_response(self,
+                                   available_bots=None,
+                                   bots_config=None,
+                                   bots_lib_module=None,
+                                   payload_url="/bots/testbot",
+                                   message=dict(message={'key': "test message"}),
+                                   check_success=False,
+                                   check_failure=False,
+                                   ):
+        # type: (List[str], Dict[str, Mapping[str, str]], Dict[str, Any], str, Dict[str, Dict[str, Any]], bool, bool) -> None
+
+        if available_bots is not None:
+            zulip.bot_server.available_bots = available_bots
+
+        if bots_config is not None:
+            zulip.bot_server.bots_config = bots_config
+
+        if bots_lib_module is not None:
+            zulip.bot_server.bots_lib_module = bots_lib_module
+
+        response = self.app.post(payload_url, data=json.dumps(message))
+
+        if check_failure:
+            assert response.status_code >= 400 and response.status_code < 500
+
+        if check_success:
+            assert response.status_code >= 200 and response.status_code < 300
```

Comment:
Instead of having `check_failure` and `check_success`, I think we can get rid of `check_failure` and do something like this for clarity:
``` python
if check_success:
     assert response.status_code >= 200 and response.status_code < 300
else:
     assert response.status_code >= 400 and response.status_code < 500
```
That would make the assertion helper function much simpler! :) This way we only have one argument. Having two separate ones is redundant when `check_success` can represent both. :)